slither/tools/read_storage/read_storage.py

```diff
@@ -42,18 +44,41 @@ class SlitherReadStorageException(Exception):
     pass
 
 
+class RpcInfo:
+    def __init__(self, rpc_url: str, block: Union[int, str] = "latest") -> None:
+        assert (
+            isinstance(block, int)
+            or block.isnumeric()
+            or block in ["latest", "earliest", "pending", "safe", "finalized"]
```

Comment:
Since we cast block to an `int` if possible perhaps it'd be best to reject `"1234"` (remove `isnumeric`) and create and Enum for the valid `str` args

Reply:
Do you know of an example elsewhere in the codebase that uses an enum like this?